### ISSUE TITLE
Using Pretrained models even if it is set to false

### ISSUE THREAD

**Issue by Adithya4720** [ultralytics/ultralytics#3582]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

I have  trained a yolo model on a custom data set of fire detection even my dataset has only fire and smoke classification only it is even detecting chairs,persons,sofas, etc.. so the second time I trained with yolov8n.yaml then also it is detecting the same So how to solve this

### Environment

YOLOv8n
OS-Google Colab

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@Adithya4720 see https://docs.ultralytics.com/yolov5/tutorials/tips_for_best_training_results for guidelines on improving your training results.

**Comment by Adithya4720** [comment]
Thankyou

**Comment by glenn-jocher** [comment]
@Adithya4720 thank you for bringing this issue to our attention. In YOLOv8, the model may still detect objects that were not included in the training dataset, even if the `pretrained` flag is set to `False`. This is because the YOLOv8 model has been trained on a diverse set of objects and has learned to detect a wide range of classes.

To address this issue and improve the model's performance on fire and smoke detection, you can try the following:
1. Ensure that your training dataset contains a sufficient number of examples for fire and smoke classes.
2. Augment the dataset with variations of fire and smoke images, such as different lighting conditions, angles, and scales.
3. Increase the number of training iterations to allow the model to learn more effectively.
4. Experiment with different hyperparameters, such as learning rate and batch size, to optimize the training process.

By following these recommendations, you can enhance the model's ability to detect fire and smoke while reducing false detections of other objects.

We appreciate your contribution to the YOLOv8 community, and if you have any further questions or need assistance, feel free to ask.

**Comment by jbdebelle** [comment]
@glenn-jocher, @Burhan-Q,

Could you confirm whether for the detection model, setting the pretrained flag to either False or True makes no difference? Will it always utilize pretrained weights regardless of this setting?

**Comment by Burhan-Q** [comment]
@jbdebelle a valid `.pt` model provided with be used as "starting" weights no matter what. The most significant use of the `pretrained` argument in the code (I'm aware of) is used for classification training:

https://github.com/ultralytics/ultralytics/blob/cb99f71728b66aefc6b03ef326c7f85968316ca2/ultralytics/models/yolo/classify/train.py#L51-L52

The other siginificant use would be when loading a model using a YAML config file, then using a `pretrained="my_custom_model.pt"` in the model `overrides` dictionary:

https://github.com/ultralytics/ultralytics/blob/cb99f71728b66aefc6b03ef326c7f85968316ca2/ultralytics/cfg/__init__.py#L557-L558 

which would attempt to load the weights into the model that was created from the YAML configuration.

**Comment by javierselva** [comment]
> [@glenn-jocher](https://github.com/glenn-jocher), [@Burhan-Q](https://github.com/Burhan-Q),
> 
> Could you confirm whether for the detection model, setting the pretrained flag to either False or True makes no difference? Will it always utilize pretrained weights regardless of this setting?


**tl;dr**
I think it makes no difference. To control whether to use a pretrained set of weights or training from scratch, you have to pass to the 'model' parameter, respectively, a '.pt' file or a '.yaml' file ([one of the files here](https://github.com/ultralytics/ultralytics/tree/main/ultralytics/cfg/models)). Passing a '.pt' file will load the pretrained weights, and passing a '.yaml' will start training from scratch. 'pretrained' parameter as a _bool_ only seems to have effect if solving a classification task.

**Long comment**
Indeed, all tasks except classification, seem to load the weights if they exist, regardless of the 'pretrained' parameter. For instance, for detection and segmentation:

[/ultralytics/ultralytics/models/yolo/detect/train.py](https://github.com/ultralytics/ultralytics/blob/cb99f71728b66aefc6b03ef326c7f85968316ca2/ultralytics/models/yolo/detect/train.py#L86C1-L92C1)

[/ultralytics/ultralytics/models/yolo/segment/train.py](https://github.com/ultralytics/ultralytics/blob/cb99f71728b66aefc6b03ef326c7f85968316ca2/ultralytics/models/yolo/segment/train.py#L32C5-L38C21)

In this sense, I'm not sure why in classification it's done differently, but that behaviour makes more sense to me. Also, I don't get why it's loading the weights, then resetting them if 'pretrained=False'... Maybe just adding 'and self.pretrained' to the if statement that checks if weights must be loaded...But I digress. I guess that, in order to be able to pass a .pt file to model and still get the desired pretrained=False behaviour one should modify these 'train.py' files to behave similarly to the classification model. 

Despite all this, these functions are called, as it's my understading, from here:

[/ultralytics/ultralytics/engine/trainer.py](https://github.com/ultralytics/ultralytics/blob/cb99f71728b66aefc6b03ef326c7f85968316ca2/ultralytics/engine/trainer.py#L525C5-L538C20)

Where the weights will be loaded from either the 'self.model' parameter or 'self.pretrained' if they contain a valid path to a '.pt' file. Note that in [/ultralytics/ultralytics/engine/trainer.py](https://github.com/ultralytics/ultralytics/blob/cb99f71728b66aefc6b03ef326c7f85968316ca2/ultralytics/engine/trainer.py#L129) '.pt' is added to 'self.model' if it contains no suffix. Now that I think of it, if 'model' is the path to a '.yaml' file and pretrained=True I don't think that will work _as expected_ either (it will train from scratch as well, as [there's no else clause](https://github.com/ultralytics/ultralytics/blob/cb99f71728b66aefc6b03ef326c7f85968316ca2/ultralytics/engine/trainer.py#L525C5-L538C20)). So... 'pretrained' is not actually doing much here. If you pass a '.pt' as model it will use those weights, and if you pass a '.yaml' file it will train from scratch. I guess that, having both parameters exist, what would make sense is to reset the model (as in classification) if a '.pt' file is provided but 'pretrained=False'.

Well, in summary, that was precisely my problem, assuming that 'pretrained' ruled over 'model'. I was calling with 'model_name.pt' and pretrained=False, hence using a pretrained model, because having '.pt' in the model name overrides pretrained=False. To actually get desired behaviour and train from scratch you have to call with 'model_name.yaml' (and maybe pretrained=False, I think it's not necessary, but I have not tested it).

**Comment by glenn-jocher** [comment]
@jbdebelle You're right to question this. For detection models, the behavior depends on how you initialize the model:

1. If you pass a `.pt` file to the `model` parameter (e.g., `model="yolo11n.pt"`), it will use those pretrained weights regardless of the `pretrained` flag.

2. If you pass a `.yaml` file (e.g., `model="yolo11n.yaml"`), it will initialize the model architecture without pretrained weights.

To train truly from scratch on only your classes:
```python
model = YOLO("yolo11n.yaml")  # Initialize model architecture only
results = model.train(data="your_dataset.yaml", epochs=100)
```

For @hegdeadithyak's original issue, this approach should help ensure the model only learns to detect fire and smoke. Make sure your dataset is sufficiently large and diverse to enable the model to learn these specific classes effectively.

**Comment by hegdeadithyak** [comment]
Yes it worked thanks for helping😁

**Comment by glenn-jocher** [comment]
Great to hear it worked for you! This is a common point of confusion with YOLO models. Using the `.yaml` configuration file instead of `.pt` ensures you're truly training from scratch without any pretrained knowledge. If you run into any other issues with your fire detection model, feel free to open a new discussion. Happy detecting!
